BlogTemplate/Pages/Edit.cshtml.cs

```diff
@@ -26,72 +23,60 @@ public EditModel(BlogDataStore dataStore, SlugGenerator slugGenerator, ExcerptGe
 
         public void OnGet([FromRoute] int id)
         {
-            newPost = oldPost = _dataStore.GetPost(id);
+            Post post = _dataStore.GetPost(id);
+
+            EditedPost = new EditedPostModel
+            {
+                Title = post.Title,
+                Body = post.Body,
+                Excerpt = post.Excerpt,
+            };
 
-            if (oldPost == null)
+            if (post == null)
             {
                 RedirectToPage("/Index");
             }
         }
 
-
         [ValidateAntiForgeryToken]
-        public IActionResult OnPostPublish([FromRoute] int id)
+        public IActionResult OnPostPublish([FromRoute] int id, [FromForm] bool updateSlug)
         {
-            oldPost = _dataStore.GetPost(id);
-            newPost.IsPublic = true;
-            UpdatePost(id);
-            return Redirect($"/Post/{id}/{newPost.Slug}");
+            Post post = _dataStore.GetPost(id);
+            post.IsPublic = true;
+            UpdatePost(post, updateSlug);
+            return Redirect($"/Post/{id}/{post.Slug}");
         }
 
         [ValidateAntiForgeryToken]
-        public IActionResult OnPostSaveDraft([FromRoute] int id)
+        public IActionResult OnPostSaveDraft([FromRoute] int id, [FromForm] bool updateSlug)
         {
-            oldPost = _dataStore.GetPost(id);
-            newPost.IsPublic = false;
-            UpdatePost(id);
-            return Redirect("/Index");
+            Post post = _dataStore.GetPost(id);
+            post.IsPublic = false;
+            UpdatePost(post, updateSlug);
+            return Redirect("/Drafts");
         }
-
-        private void UpdatePost(int id)
+        private void UpdatePost(Post post, [FromForm] bool updateSlug)
         {
-            newPost.Id = id;
-            oldPost = _dataStore.GetPost(id);
-
-            if (oldPost.PubDate.Equals(default(DateTime)))
-            {
-                if (newPost.IsPublic == true)
-                {
-                    newPost.PubDate = DateTime.UtcNow;
-                }
-            }
-            else
-            {
-                newPost.PubDate = oldPost.PubDate;
-            }
-
-            if (newPost.Excerpt == null)
-            {
-                newPost.Excerpt = _excerptGenerator.CreateExcerpt(newPost.Body, 140);
-            }
+            post.Title = EditedPost.Title;
+            post.Body = EditedPost.Body;
+            post.Excerpt = EditedPost.Excerpt;
 
-            if (Request.Form["updateslug"] == "true")
-            {
-                newPost.Slug = _slugGenerator.CreateSlug(newPost.Title);
-            }
-            else
+            _dataStore.SavePost(post);
+            if (updateSlug)
             {
-                newPost.Slug = oldPost.Slug;
+                post.Slug = _slugGenerator.CreateSlug(post.Title);
             }
-            newPost.Comments = oldPost.Comments;
+        }
 
-            _dataStore.UpdatePost(newPost, oldPost);

```

Comment:
The issue with using _dataStore.SavePost instead of _dataStore.UpdatePost is that it doesn't delete the old file. For example, when I publish a post, edit the post, and save it as a draft, it still shows up on the index page because the old file under BlogFiles\Posts still exists.